namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStore.scala

```diff
@@ -0,0 +1,106 @@
+package io.buoyant.namerd
+package storage.consul
+
+import com.twitter.finagle.{Dtab, Path}
+import com.twitter.io.Buf
+import com.twitter.util._
+import io.buoyant.consul.v1._
+
+import io.buoyant.namerd.DtabStore.{DtabNamespaceAlreadyExistsException, DtabNamespaceDoesNotExistException}
+
+class ConsulDtabStore(api: KvApi, root: Path) extends DtabStore {
+  import DtabStore.{DtabVersionMismatchException, Version}
+
+  override def list(): Activity[Set[Ns]] = {
+    def namespace(key: String): Ns = key.stripPrefix("/").stripSuffix("/").substring(root.show.length)
+
+    val run = Var.async[Activity.State[Set[Ns]]](Activity.Pending) { updates =>
+      @volatile var running = true
+
+      def cycle(index: Option[String]): Future[Unit] =
+        if (running)
+          api.list(s"${root.show}/", blockingIndex = index) transform {
+            case Return(result) =>
+              val namespaces = result.value.map(namespace).toSet
+              updates() = Activity.Ok(namespaces)
+              cycle(result.index)
+            case Throw(e: NotFound) =>
+              updates() = Activity.Ok(Set.empty[Ns])
+              cycle(e.rsp.headerMap.get(Headers.Index))
+            case Throw(e) =>
+              updates() = Activity.Failed(e)
+              cycle(None)
+          }
+        else
+          Future.Unit
+      val pending = cycle(None)
+
+      Closable make { _ =>
+        running = false
+        pending.raise(new FutureCancelledException)
+        Future.Unit
+      }
+    }
+
+    Activity(run)
+  }
+
+  def create(ns: Ns, dtab: Dtab): Future[Unit] =
+    api.put(s"${root.show}/$ns", dtab.show, cas = Some("0")) flatMap { result =>
+      if (result) Future.Done else Future.exception(new DtabNamespaceAlreadyExistsException(ns))
+    }
+
+  def delete(ns: Ns): Future[Unit] = {
+    val key = s"${root.show}/$ns"
+    api.get(key) transform {
+      case Return(_) => api.delete(key).unit
+      case Throw(e: NotFound) => Future.exception(new DtabNamespaceDoesNotExistException(ns))
```

Comment:
as far as I can tell, consul seems to always return 200 and "true", even if the key doesn't exist.  *shrug*